### DISCUSSION TITLE
Unable to resume after internet connection was interrupted for a few seconds

### DISCUSSION THREAD

**Discussion by lamyergeier** [aclap-dev/video-downloadhelper#836] [Help and Q&A]
### Your question

Unable to resume after internet connection was interrupted for a few seconds, the GUI shows that the download is progressing but the percentage of downloaded content remains same and won't change.

Is the extension supports resume after internet connection was interrupted? If not, what alternatives do I have? Will a Download manager work better?

### Addon & CoApp version and details

Version 9.0.2.3 (stable), target mozilla, locale en-US.

Companion App installed (v2.0.19)

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I am just another user.

I have occasionally seen VDH pause a download for several minutes, perhaps as long as 5 minutes or longer.  That seems like it would be long enough to cause the entire download to time out & you would expect VDH to say it has successfully completed the download when in fact it just stopped prematurely.  But I have also seen it resume successfully & eventually complete successfully with the complete video residing on my HDD.  It just takes some patience.  If you give us some more detail, like some sample URLs on which you are observing this behavior, I might be able to observe this same behavior.

On the other hand, in cases where I have observed these disturbing pauses, I have also seen that if I try again a while later, several hours or even the next day, the problem does not recur.  It seems to be a transient problem with the serving web site.  So there's only a chance, not a certainty, that I would observe the same thing you observe.  That's not to deny that you are seeing a problem.  I have seen the same problem.  But it's highly unpredictable, inconsistent, not always repeatable.

Patience, even unreasonable patience, might be required here.

**Reply by GrampaWildWilly** [reply]
On the other hand, if your Internet connection has actually disconnected, not simply paused, you may be out of luck.  VDH does not have a feature to support resumption of a failed download from the point of failure.  This was a discussion I had with Michel a few years ago.  It seems there are a lot of complicating factors involved, especially when we're talking about HLS & DASH streams, as opposed to simple files.  HLS & DASH don't transmit files.  They transmit sections of files as if they were separate files.  When you receive all the sections, it looks like it was just one file.  But under the covers, the file is chopped up into chunks.  According to Michel, this makes resumption from a point of failure impossible.

Who is Michel?  He is Michel Gutierrez, the originator of VDH.  He used to post as mig on our old & apparently no longer available forum on Google Groups.  He has mostly turned development of the product over to @paulrouget Paul since about a year ago.  I haven't seen Michel posting in this forum yet.  It would be a most welcome sight.

But back to your issue.  After a complete loss of Internet connectivity, even if only momentary, you're pretty much stuck.  In such a case, you should probably reload the extension.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/76de755d-cec9-4617-9877-295d6f1f00b6)
![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/062910e8-5ed3-4709-a9f5-e722117216b4)
![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/77142737-7d7f-4396-981e-1d28285f1054)

This will clear the download that is stuck in VDH.  Simply clearing the entry off the menu by clicking its little x above it only removes the entry from the menu.  It does not trash the zombie download.  Or downloads, plural.  It is possible to have several downloads in this zombie state.  Only reloading the extension does the trick.  (Or closing all browser windows & starting over, but that's a bit extreme & probably very inconvenient.)  However, that also clears any waiting downloads you might have in your queue.  They were probably doomed by the disconnection anyway.  But you can see that this means every download must be started over from the beginning.  Any partial files you already have will be useless & should be discarded or just replaced by your restarted downloads.  This means that it is always a prudent practice to keep open any web page from which you have started a download.  Don't close the page until you verify that your download has completed successfully.  VDH doesn't require you to keep a page open while a download from that page is in progress.  But because of precisely the situation you are describing, it is a smart move to keep all your relevant pages open until you have manually verified that VDH has done its job on all your downloads.
